test/recipes/fuzz.pl

```diff
@@ -9,23 +9,20 @@
 use warnings;
 
 use OpenSSL::Glob;
-use OpenSSL::Test qw/:DEFAULT srctop_file/;
+use OpenSSL::Test qw/:DEFAULT srctop_dir/;
 
-sub fuzz_tests {
-    my @fuzzers = @_;
+sub fuzz_test {
+    die "Only one argument accepted" if scalar @_ != 1;
 
-    foreach my $f (@fuzzers) {
-        subtest "Fuzzing $f" => sub {
-            my @dir = glob(srctop_file('fuzz', 'corpora', "$f"));
+    my $f = $_[0];
+    my $d = srctop_dir('fuzz', 'corpora', $f);
 
-            plan skip_all => "No directory fuzz/corpora/$f" unless @dir;
-            plan tests => scalar @dir; # likely 1
+    return run(fuzz(["$f-test", $d])) if -d $d;
 
-            foreach (@dir) {
-                ok(run(fuzz(["$f-test", $_])));
-            }
-        }
-    }
+    # Directory $d doesn't exist if we reach this point, so write a
+    # diagnostic and return false.
+    diag("No directory $d");
+    return 0;
 }
```

Comment:
With the latest change in the recipes, this function is becoming pointless...  Another way to restore elegance could be this rewrite:

``` perl
sub fuzz_ok ($;$) {
    my $f = $_[0];
    my $title = $_[1];
    my $d = srctop_dir('fuzz', 'corpora', $f);
  SKIP: {
      skip "No directory $d", 1 unless -d $d;
      ok(run(fuzz(["$f-test", $d])), $title);
    }
}
```

The call in the recipes would then be something like:

``` perl
fuzz_ok($fuzzer, "Fuzzing $fuzzer");
```

Reply:
What about f1f4a3c3468ae07b2eb1c0dc3023a21440ca30a0 ?